### PR TITLE
Protect: Add WAF StatCards

### PR DESCRIPTION
**Pull request by dkmyta**
## Proposed changes:

* Makes the `StatCard` js-packages component easily accessible
* Adds `Waf_Blocklog_Manager` class and migrates related methods over 
* Adds a new set of methods for tracking daily and monthly blocked requests stats via new `jetpack_waf_blocklog_daily_summary`,  and `jetpack_waf_all_time_stats` options
* Adds the `StatCard` components to the `FirewallHeader` and hooks up the new stats

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

- https://github.com/Automattic/jetpack-scan-team/issues/381

## Does this pull request change what data or activity we track or use?

- No

## Testing instructions:

* Check out this branch and start up your Jurassic Tube with Protect installed and active with a scan-level plan
* Enable the WAF, ensure share data is toggled, configure your settings as desired
* Register several blocked requests and verify that the stats in the UI are updated accordingly on refresh
* Modify the `$date` value in the `update_daily_summary()` method of the `Waf_Blocklog_Manager` class to simulate requests from a different day within the same month and from outside of the current month
* Verify that the `jetpack_waf_blocklog_daily_summary`, and `jetpack_waf_all_time_stats` option data and stats in the UI reflect the entries and that any data logged outside of the current month is pruned
* Verify that stats are 0 and disabled when a scan upgrade is removed (paid feature flag is set) and/or share data is disabled
* Ensure `jetpack-waf/waf-blocklog` entries or `jetpack_waf_blocklog_daily_summary`, and `jetpack_waf_all_time_stats` option updates are made when share data is disabled (note that `wp_jetpack_waf_blocklog` table entries will continue as well)
* Modify the option data to verify display when values are significant
* Switch between desktop and mobile view to ensure the display is responsive
* Ensure no regressions are introduced relating to WAF and blocklog operations
* Run `cd projects/js-packages/storybook && npm run storybook:dev` and verify that all Protect `FirewallHeader` stories render
* Run `jetpack test packages/waf php` to ensure that all integration tests pass

## Screenshots:
Desktop:
Screen Shot 2024-02-22 at 15 07 29

Mobile:
Screen Shot 2024-02-22 at 15 08 15

### CODE REVIEW

File: projects/packages/waf/.phan/baseline.php
Comment (nateweller) [inline projects/packages/waf/.phan/baseline.php:1]: Are these automated or manual changes to this file? Are they necessary?

**Review by nateweller — DISMISSED** [review]

File: projects/packages/waf/.phan/baseline.php
Comment (dkmyta) [reply projects/packages/waf/.phan/baseline.php:1]: Good question 🤔 I believe they were necessary at some point to satisfy some static analysis errors and were either manually applied after reviewing the output of those tests or automatically applied by running the script to update phan baseline. Lots of changes have been made between now and then though, so its possible they are no longer necessary. I'll revert and see if they crop up again.

**UPDATE**: Removed the changes and the errors reappeared - reapplied to resolve.

File: projects/packages/waf/src/class-waf-blocklog-manager.php [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+		}
+	}
+
+	/**
+	 * Increment the all-time stats.
+	 *
+	 * @return void
+	 */
+	private static function write_all_time_block_count() {
+		$block_count = self::get_all_time_block_count_value() ?? self::get_default_all_time_stat_value();
```
Comment (nateweller) [inline projects/packages/waf/src/class-waf-blocklog-manager.php]:
```suggestion
		$block_count = self::get_all_time_block_count_value();
		if ( ! $block_count ) {
			$block_count = self::get_default_all_time_stat_value();
		}
```
Whenever the block count is zero, use the default value.

**Review by nateweller — APPROVED** [review]
LGTM! :+1:

**Pull request merged**